osquery/database/database.cpp

```diff
@@ -343,4 +350,51 @@ void dumpDatabase() {
     }
   }
 }
+
+Status updateDatabase() {
+  std::vector<std::string> keys;
+  auto s = scanDatabaseKeys(kQueries, keys);
+  if (!s.ok()) {
+    return Status(1, "Failed to lookup legacy query data from database");
+  }
+
+  for (const auto& key : keys) {
+    std::string value;
+    if (!getDatabaseValue(kQueries, key, value)) {
+      LOG(WARNING) << "Failed to get value from database " << key << ": "
+                   << value;
+      continue;
+    }
+
+    pt::ptree tree;
+    try {
+      std::stringstream ss;
+      ss << value;
+      pt::read_json(ss, tree);
+    } catch (const pt::json_parser::json_parser_error& /* e */) {
+      LOG(WARNING) << "Conversion from ptree to RapidJSON failed for " << key
+                   << ": " << value;
+    }
+
+    auto json = JSON::newArray();
+    for (const auto& t : tree) {
+      std::stringstream ss;
+      pt::write_json(ss, t.second);
+
+      rj::Document row;
+      if (row.Parse(ss.str()).HasParseError()) {
+        LOG(WARNING) << "Failed to serialize JSON row for " << key;
+      }
+      json.push(row);
+    }
+
+    std::string out;
+    json.toString(out);
+    if (!setDatabaseValue(kQueries, key, out)) {
+      LOG(WARNING) << "Failed to update value in database " << key << ": "
+                   << value;
+    }
+  }
+  return Status();
```

Comment:
Nit Status(0)

Reply:
I think `Status()` looks better, what do you guys think? The default `Status()` ctor is a success.

Reply:
I'd prefer `Status()`, I think it's cleaner and I know we've been moving to that in other places in the code base.